packages/editor/src/browser/editor-preferences.ts

```diff
@@ -77,6 +77,11 @@ const codeEditorPreferenceProperties = {
         'minimum': 1,
         'markdownDescription': 'The number of spaces a tab is equal to. This setting is overridden based on the file contents when `#editor.detectIndentation#` is on.'
     },
+    'editor.defaultFormatter': {
+        'type': 'string',
+        'default': 'default',
+        'description': 'Default formatter'
+    },
```

Comment:
Is there a reason we do not want to support this same schema as vscode, especially the default value (`null`)?
I believe it will cause inconsistencies:

https://github.com/microsoft/vscode/blob/68410abc36030d3fb321145d2e4945f72f6e6bee/src/vs/workbench/contrib/format/browser/formatActionsMultiple.ts#L157-L164

Reply:
PR is updated. For default value I used `undefined`, as we cannot use `null`. Thanks
```
lerna ERR! /home/vitaliy/projects/vitaliy-guliy-theia/packages/editor/src/browser/editor-preferences.ts
lerna ERR!   82:20  error  Use undefined instead of null  no-null/no-null
```

Reply:
We could use it if we disable the rule for it, I think in this case it is fine since we want compatibility:

```            
// eslint-disable-next-line no-null/no-null
```

My only concern is when reading `.vscode` settings where they use `null` as a value since it is allowed.